modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/utils/ScriptsGitRepoManage.java

```diff
@@ -39,7 +39,7 @@ public class ScriptsGitRepoManage {
     @Retryable(retryFor = GitRepoCloneException.class,
             maxAttemptsExpression = "${http.request.retry.max.attempts}",
             backoff = @Backoff(delayExpression = "${http.request.retry.delay.milliseconds}"))
-    public void checkoutScripts(String workspace, ScriptsRepo scriptsRepo) {
+    public File[] checkoutScripts(String workspace, ScriptsRepo scriptsRepo) {
```

Comment:
this seems to be duplicate work. 
The file is already available in the GIT checkout. We are loading it into memory and writing new files again. 
We must try to directly run the HCLParser from the GIT repo location.